include/threads.h

```diff
@@ -26,9 +26,23 @@
 #ifndef _THREADS_H_
 #define _THREADS_H_
 
-void logging_thread_cleanup(void *arg);
+#include <pthread.h>
+#include <unistd.h>
 
 void *logging_thread(void *arg);
 void *reading_thread(void *arg);
 
+// see https://blog.memzero.de/pthread-cancel-noexcept/
+
+#define CANCELLABLE(operation)                                                                     \
+	{                                                                                              \
+		pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);                                       \
+		operation;                                                                                 \
+		pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, NULL);                                      \
```

Comment:
would be nicer if you'd restore the old state (pass a ptr to _setcancelstate instead of the NULL and use that value)

Reply:
Thanks for the tip!